tc/core/polyhedral/cuda/mapped_scop.cc

```diff
@@ -587,6 +575,22 @@ std::tuple<std::string, tc::Grid, tc::Block> MappedScop::codegen(
       mappedScopForCodegen->numThreads);
 }
 
+// Split out reduction loops into separate bands and insert reduction
+// synchronizations after those bands.
+void MappedScop::insertReductionSyncs() {
```

Comment:
Maybe rename the method to reflect the two operation?
`splitOutReductionsAndInsertSyncs`?